File: FifaDataSim/Models/KnockoutMatch.cs
namespace WorldCupSimulator.Models;

public class KnockoutMatch
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Stage { get; set; } = string.Empty; // "R32", "R16", "QF", "SF", "Final"
    public int MatchNumber { get; set; }
    
    public Country? HomeTeam { get; set; }
    public Country? AwayTeam { get; set; }
    
    public int? HomeScore { get; set; }
    public int? AwayScore { get; set; }
    public bool WentToExtraTime { get; set; }
    public int? HomeExtraTimeScore { get; set; }
    public int? AwayExtraTimeScore { get; set; }
    public bool WentToPenalties { get; set; }
    public int? HomePenaltyScore { get; set; }
    public int? AwayPenaltyScore { get; set; }
    
    public Country? Winner { get; set; }
    public bool IsPlayed => Winner != null;
}

File: FifaDataSim.React/src/types/knockoutMatch.ts
import type { Country } from "./country";

export interface KnockoutMatch {
  id: string;
  stage: 'R32' | 'R16' | 'QF' | 'SF' | 'ThirdPlace' | 'Final';
  matchNumber: number;
  homeTeam: Country | null;
  awayTeam: Country | null;
  homeScore: number | null;
  awayScore: number | null;
  wentToExtraTime: boolean;
  homeExtraTimeScore: number | null;
  awayExtraTimeScore: number | null;
  wentToPenalties: boolean;
  homePenaltyScore: number | null;
  awayPenaltyScore: number | null;
  winner: Country | null;
  isPlayed: boolean;
}